docs/plugin_developers_guide/07_rules_plugins.rst

```diff
@@ -0,0 +1,432 @@
+            rule->setState(eval);
+
+            return eval;
+    }
+
+In this case the code iterates through the trigger names and calls the *evaluate* method in the *Average* class for each trigger and with each value in the incoming data stream.
+
+Various calls are made that will set the state of the evaluation, namely the *setValTimestamp* and *setState*. These states may later be used in the *plugin_reason* API.
+
+Plugin Reason
+~~~~~~~~~~~~~
+
+The *plugin_reason* API call is made to the rule plugin by the notification service when it needs to send a notification. Depending on the result of the last *plugin_evaluate* call this may a notification of the condition either triggering or clearing. The code for the Moving Average plugin is shown below.
+
+.. code-block:: C
+
+    /**
+     * Return rule trigger reason: trigger or clear the notification. 
+     *
+     * @return	 A JSON string
+     */
+    string plugin_reason(PLUGIN_HANDLE handle)
+    {
+            AverageRule* rule = (AverageRule *)handle;
+            BuiltinRule::TriggerInfo info;
+            rule->getFullState(info);
+
+            string ret = "{ \"reason\": \"";
+            ret += info.getState() == BuiltinRule::StateTriggered ? "triggered" : "cleared";
+            ret += "\"";
+            ret += ", \"asset\": " + info.getAssets();
+            if (rule->getEvalTimestamp())
+            {
+                    ret += string(", \"timestamp\": \"") + info.getUTCTimestamp() + string("\"");
+            }
+            ret += " }";
+
+            return ret;
+    }
+
+The code first fetches the state information that was set by the previous *plugin_eval* call, using the *getFullState()* entry point of the base *BuiltinRule* class.
+
+The *plugin_reason* call returns a JSON document, within a string. The reason document returns why the notification is being sent, the name of the item that triggered the notification and the timestamp of the data that triggered the notification. Below is a example reason document.
+
+.. code-block:: JSON
+
+   {
+       "reason" : "triggered",
+       "asset" : "sinusoid",
+       "timestamp" : "2025/03/16 12:33:04.026"
+   }
+
+.. note::
+
+   The data that triggered the notification is always passed with a key of *asset*, but it may be an asset in a reading, a statistic name or an audit log code.
+
+Plugin Reconfigure
+~~~~~~~~~~~~~~~~~~
+
+As with other plugin types the notification delivery plugin  may be
+reconfigured during its lifetime. When a reconfiguration operation occurs
+the *plugin_reconfigure* method will be called with the new configuration
+for the plugin.
+
+.. code-block:: C
+
+   void plugin_reconfigure(PLUGIN_HANDLE *handle, const std::string& newConfig)
+   {
+        AverageRule *average = (AverageRule *)handle;
+        average->configure(newConfig);
+        return;
+   }
+
+In the case of the Moving Average plugin this calls the same *configure* method that was called in the *plugin_init* call and is shown above.
```

Comment:
this invokes the same *configure* method that was previously called during the *plugin_init* invocation, as shown above.